### PR TITLE
fix(sidebar): open menu, highlight current page

### LINKED ISSUE #5547: Suggestion: Highlight current article in the left sidebar
MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Tutorial_local_library_website

I have been using MDN since the beginning of my programming journey. It has been a main online source for learning web technologies and looking up things that I forget. I am also so pleased to see the new design of the MDN website. It is stylish and feels nice.

Here I am not intending to report a problem or an issue. It is a suggestion to improve the user interface.
image
I am on [this page](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Tutorial_local_library_website). There are a list of navigation links on the **left** sidebar. But I can not detect which page I am viewing by looking at those links on the left. That would be useful to see the current page link having a little apparent style among other links.

Thank you

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/learn/server-side/express_nodejs/tutorial_local_library_website`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Tutorial_local_library_website
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/server-side/express_nodejs/tutorial_local_library_website/index.md
* Last commit: https://github.com/mdn/content/commit/0b270d4f95ba1c4abcc81215df774a9448a78105
* Document last modified: 2022-02-18T09:10:31.000Z

</details>

### LINKED ISSUE #5641: Left sidebar navigation collapses on CSS reference pages
### Summary

The menus under "Reference" and "Guides" options on the left sidebar collapse after clicking a link from the sidebar in CSS Reference pages. 

### URL

https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors

or any [Web/CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) page

### Reproduction steps

1. "References" > "CSS"
2. Left sidebar: Click "Properties" under **`Reference`** to display the drop-down menu
3. Click "align-content".
 The menu on left collapses

1. "References" > "CSS"
2. Left sidebar: Click "Box model" under **`Guides`** to display the drop-down menu
3. Click "Mastering margin collapsing".
 The menu on left collapses

### Expected behavior

- The menu to stay expanded
- Highlight the current page in the sidebar (this is being discussed here: https://github.com/mdn/yari/issues/5547)

### Actual behavior

The menu collapses

### Device

Desktop

### Browser

Firefox

### Browser version

Stable

### Operating system

Mac OS

### Screenshot

Screenshot 2022-03-16 at 12 54 34 PM

### Anything else?

_No response_

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mdn/yari/discussions).

### PR DESCRIPTION
**Pull request by caugner**
## Summary

Highlights the current page in the sidebar, and expands the corresponding menu.

Fixes #5547.
Fixes #5641.

### Problem

1. The current page is not highlighted in the sidebar.
2. When opening a page from an expanded menu, the menu is collapsed again.

### Solution

1. Mark the current page with [aria-current](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current).
2. Open `<details>` elements that contain links to the current page.

---

## Screenshots

| Before | After |
|-------|-------|
| image | image  |

---

## How did you test this change?

Opened these pages locally:

- http://localhost:3000/en-US/docs/Web/CSS/CSS_Animations
- http://localhost:3000/en-US/docs/Web/JavaScript/Reference/Operators/Assignment
- http://localhost:3000/en-US/docs/Web/HTML/Global_attributes/class
- http://localhost:3000/en-US/docs/Web/HTTP/CSP
- http://localhost:3000/en-US/docs/Learn/CSS/Styling_text/Fundamentals
- http://localhost:3000/en-US/docs/Learn/HTML/Introduction_to_HTML/Creating_hyperlinks
- http://localhost:3000/en-US/docs/Learn/JavaScript/Building_blocks/conditionals

### CODE REVIEW

**Comment by wbamberg** [comment]
This is great to see! cc @hamishwillee who had some good comments at https://github.com/mdn/yari/issues/5547#issuecomment-1067567913 .

It would be even better if we would scroll the sidebar so the highlighted link is visible. Is that planned as a follow-up?

A few suggestions.

1. I think this sort of thing works better when the highlight takes up the full width of the sidebar, so all highlights are the same width, whatever the length of the link title:

Screen Shot 2022-09-07 at 4 08 04 PM

2. It would be nice if the highlight in the sidebar used the same highlighting as the highlight-current-section in the ToC:

Screen Shot 2022-09-07 at 4 09 23 PM

As it is things look inconsistent.

3. The current highlighting has poor contrast: https://contrastchecker.com/?c=cdcdcd&b=696969.

**Comment by caugner** [comment]
@wbamberg Good catch, and the highlighted headings didn't look good either with their padding, so I updated the strategy a bit, wrapping the links **and their siblings** in an `<em>` like the JSRef macro already used to do.

I have updated the screenshot in the PR description.

**Comment by wbamberg** [comment]
I think this is much better, thank you! The border and left-padding indents the current page entry:

Screen Shot 2022-09-08 at 4 30 40 PM

...this is a pretty nit-picky detail though so I won't insist on it :).

**Review by wbamberg — APPROVED** [review]
Approving but I'm happy for you to adjust things so the highlighted item is not indented, if you think it is worth doing. Thanks @caugner !

**Review by hamishwillee — APPROVED** [review]
@caugner Thank you. This is excellent - I've been complaining about how crappy our sidebar is for such a long time. I'd actually given up.

This is vastly better than the current situation. Tested:
- Opening a page by copying an a URL collapses the sidebar except for the section with the page, and highlights the page in sidebar. Excellent.
- Opening a page from sidebar does the same thing.
- - Opening a page from a link does the same thing.
- Sidebar highlight is clearly visible in all sections and easy to see if you scroll down the sidebar.
- Sidebar highlight good in both light and dark theme.

A few comments:

- Nothing is highlighted in sidebar for a top level entry point like http://localhost:3000/en-US/docs/Web/CSS . Should it be? Because I expected a sidebar entry I was looking.
- The different colour highlighting in different areas of the site do no harm, but add not benefit (IMO). I could be wrong for others.
- You can navigate the sidebar with a page open and the tree remain open until you select a new page. This is valid and fine. Note though that some people prefer that if you open one tree you close all others, even when "browsing" (I am one of those).
- The only flaw is that the sidebar doesn't scroll to visibility on opening - which you plan on addressing in a separate issue. 

I am approving this. Happy for it to be merged **right now**.

**Comment by hamishwillee** [comment]
@wbamberg Thanks for pinging me in. FWIW Without scrolling to the current page I still don't think we can dump the Learn "QuickLinks". The current page is findable now, but you don't have the easy context: (from http://localhost:3000/en-US/docs/Learn/Server-side/Django/Authentication#in_this_module)

When we get to that point, I wonder if we should put the tutorial in its own "folder"? 
image

**Review by Rumyra — APPROVED** [review]

File: build/document-extractor.ts [hunk 1]
```diff
@@ -36,14 +37,25 @@ type SectionsAndFlaws = [Section[], string[]];
...
  * ...give or take some whitespace.
  */
-export function extractSidebar($: cheerio.CheerioAPI) {
+export function extractSidebar($: cheerio.CheerioAPI, doc: Partial<Doc>) {
   const search = $("#Quick_links");
+
   if (!search.length) {
-    return "";
+    doc.sidebarHTML = "";
   }
```
Comment (caugner) [inline build/document-extractor.ts:45]:
```suggestion
  }
  return;
```

File: build/document-extractor.ts [hunk 2]
```diff
@@ -36,14 +37,25 @@ type SectionsAndFlaws = [Section[], string[]];
  *
  * ...give or take some whitespace.
  */
-export function extractSidebar($: cheerio.CheerioAPI) {
+export function extractSidebar($: cheerio.CheerioAPI, doc: Partial<Doc>) {
   const search = $("#Quick_links");
+
   if (!search.length) {
-    return "";
+    doc.sidebarHTML = "";
```
Comment (caugner) [inline build/document-extractor.ts:44]:
```suggestion
    doc.sidebarHTML = "";
    return;
```

**Pull request merged**